### PR TITLE
Update istio to 1.21.0

### PR DESCRIPTION
**Pull request by barchw**
**Description**

Changes proposed in this pull request:

- Update istio to 1.21.0

**Related issues**
- https://github.com/kyma-project/istio/issues/671

### CODE REVIEW

File: tests/integration/features/istio/evaluation/istio_installation_demo.feature [hunk 1]
```diff
@@ -9,13 +9,13 @@ Feature: Installing Istio module with evaluation profile
     When Istio CR "istio-sample" from "istio_cr_template" is applied in namespace "kyma-system"
     Then Istio CR "istio-sample" in namespace "kyma-system" has status "Ready"
     And Istio CR "istio-sample" in namespace "kyma-system" has condition with reason "ReconcileSucceeded" of type "Ready" and status "True"
-    Then "proxy" has "requests" set to cpu - "10m" and memory - "32Mi"
-    And "proxy" has "limits" set to cpu - "250m" and memory - "254Mi"
-    And "ingress-gateway" has "requests" set to cpu - "10m" and memory - "32Mi"
+    #Then "proxy" has "requests" set to cpu - "10m" and memory - "32Mi"
```
Comment (barchw) [inline tests/integration/features/istio/evaluation/istio_installation_demo.feature:12]: Int tests for egress and proxy - follow up

File: internal/restarter/ingress_gateway.go [hunk 2]
```diff
@@ -98,8 +99,18 @@ func RestartIngressGateway(ctx context.Context, k8sClient client.Client) error {
...
 	}
+
+	patch := client.StrategicMergeFrom((&deployment).DeepCopy())
 	deployment.Spec.Template.Annotations = annotations.AddRestartAnnotation(deployment.Spec.Template.Annotations)
-	err = k8sClient.Update(ctx, &deployment)
+
+	err = retry.RetryOnError(retry.DefaultRetry, func() error {
+		err = k8sClient.Patch(ctx, &deployment, patch)
+		if err != nil {
+			ctrl.Log.Info("Retrying ingress gateway restart")
```
Comment (barchw) [inline internal/restarter/ingress_gateway.go:109]: Follow up - logging levels

**Review by kolodziejczak — CHANGES_REQUESTED** [review]

File: tests/integration/steps/destination_rule.go [hunk 3]
```diff
@@ -4,8 +4,8 @@ import (
 	"context"
 	"github.com/avast/retry-go"
 	"github.com/kyma-project/istio/operator/tests/integration/testcontext"
-	networkingv1beta1 "istio.io/api/networking/v1beta1"
-	"istio.io/client-go/pkg/apis/networking/v1beta1"
+	networkingv1alpha3 "istio.io/api/networking/v1alpha3"
+	"istio.io/client-go/pkg/apis/networking/v1alpha3"
```
[the remark this answers is not in the document]
Comment (barchw) [reply tests/integration/steps/destination_rule.go:8]: We should use the same, latest version in all our packages, all the others use v1alpha3

File: tests/integration/steps/istio.go [hunk 4]
```diff
@@ -108,80 +116,74 @@ func UninstallIstio(ctx context.Context) error {
...
+		//			err := mapstructure.Decode(operator.Spec.Components.EgressGateways[0].K8S.Resources.Limits, &res)
+		//			if err != nil {
+		//				return nil, err
+		//			}
+		//		} else {
+		//			err := mapstructure.Decode(operator.Spec.Components.EgressGateways[0].K8S.Resources.Requests, &res)
+		//			if err != nil {
+		//				return nil, err
+		//			}
+		//		}
```
[the remark this answers is not in the document]
Comment (barchw) [reply tests/integration/steps/istio.go:170]: This one yes

File: tests/integration/features/istio/evaluation/istio_installation_demo.feature (hunk 1 above)
Comment (barchw) [reply tests/integration/features/istio/evaluation/istio_installation_demo.feature:12]: No, this should stay, as the tests should be implemented in a follow up.
This feature should not change, only the implementation

File: internal/istiooperator/istio-operator-light.yaml [hunk 5]
```diff
@@ -186,6 +186,7 @@ spec:
       validationURL: ""
     cni:
       repair:
+        repairPods: false
```
Comment (barchw) [inline internal/istiooperator/istio-operator-light.yaml:189]: The repair pods need to be false, and delete true until https://github.com/istio/istio/commit/678762f4614c86e316612ffa8ac9d0b42acfce35 becomes available in a release. Without it, repair fails

File: internal/resources/controlled_resources_list.yaml [hunk 6]
```diff
@@ -25,13 +25,6 @@ resources:
       group: networking.istio.io
       version: v1alpha3
       kind: Gateway
-  - GroupVersionKind:
```
Comment (barchw) [inline internal/resources/controlled_resources_list.yaml:28]: Create a follow up card
TODO: Validate that it is not blocking deletion after upgrade from 1.20.3

Comment (barchw) [reply internal/resources/controlled_resources_list.yaml:28]: Verified TODO, looks good

File: docs/release-notes/1.5.0.md [hunk 7]
```diff
@@ -1,3 +1,9 @@
 ## New Features
 
+- Update the Istio version to 1.21.0 [#676](https://github.com/kyma-project/istio/pull/676). Read [Istio 1.21.0 Release Announcement](https://istio.io/latest/news/releases/1.21.x/announcing-1.21/) and [Change Notes](https://istio.io/latest/news/releases/1.21.x/announcing-1.21/change-notes/) for more details.
 - The version details for Istio installation are now only specified in the files located in the [`istiooperator`](https://github.com/kyma-project/istio/blob/main/internal/istiooperator) directory [#664](https://github.com/kyma-project/istio/pull/664).
+
+## Breaking Changes in Istio upstream
+
+- Istio no longer creates `IstioOperator/installed-state` CustomResource after installation. See the upstream [Istio Issue](https://github.com/istio/istio/issues/44814) and [Pull Request](https://github.com/istio/istio/pull/47860) for more information.
+- Discontinue support for `Referer` headers containing user info or fragment components. The new sanitization behaviour is described [here](https://www.envoyproxy.io/docs/envoy/v1.26.7/configuration/http/http_conn_man/headers#config-http-conn-man-headers-referer).
```
Comment (nataliasitko) [inline docs/release-notes/1.5.0.md:9]:
```suggestion
- Discontinue support for `Referer` headers containing user info or fragment components. The new sanitization behavior is described in the [Envoy's documentation](https://www.envoyproxy.io/docs/envoy/v1.26.7/configuration/http/http_conn_man/headers#config-http-conn-man-headers-referer).
```

File: docs/release-notes/1.5.0.md [hunk 8]
```diff
@@ -1,3 +1,9 @@
 ## New Features
 
+- Update the Istio version to 1.21.0 [#676](https://github.com/kyma-project/istio/pull/676). Read [Istio 1.21.0 Release Announcement](https://istio.io/latest/news/releases/1.21.x/announcing-1.21/) and [Change Notes](https://istio.io/latest/news/releases/1.21.x/announcing-1.21/change-notes/) for more details.
 - The version details for Istio installation are now only specified in the files located in the [`istiooperator`](https://github.com/kyma-project/istio/blob/main/internal/istiooperator) directory [#664](https://github.com/kyma-project/istio/pull/664).
+
+## Breaking Changes in Istio upstream
+
+- Istio no longer creates `IstioOperator/installed-state` CustomResource after installation. See the upstream [Istio Issue](https://github.com/istio/istio/issues/44814) and [Pull Request](https://github.com/istio/istio/pull/47860) for more information.
```
Comment (nataliasitko) [inline docs/release-notes/1.5.0.md:8]:
```suggestion
- Istio no longer creates `IstioOperator/installed-state` CustomResource after installation. See the upstream [Istio issue](https://github.com/istio/istio/issues/44814) and the [PR](https://github.com/istio/istio/pull/47860) for more information.
```

File: docs/release-notes/1.5.0.md [hunk 9]
```diff
@@ -1,3 +1,9 @@
 ## New Features
 
+- Update the Istio version to 1.21.0 [#676](https://github.com/kyma-project/istio/pull/676). Read [Istio 1.21.0 Release Announcement](https://istio.io/latest/news/releases/1.21.x/announcing-1.21/) and [Change Notes](https://istio.io/latest/news/releases/1.21.x/announcing-1.21/change-notes/) for more details.
 - The version details for Istio installation are now only specified in the files located in the [`istiooperator`](https://github.com/kyma-project/istio/blob/main/internal/istiooperator) directory [#664](https://github.com/kyma-project/istio/pull/664).
+
+## Breaking Changes in Istio upstream
```
Comment (nataliasitko) [inline docs/release-notes/1.5.0.md:6]:
```suggestion
## Breaking Changes in Istio Upstream
```

**Review by kolodziejczak — DISMISSED** [review]

File: docs/release-notes/1.5.0.md [hunk 10]
```diff
@@ -1,3 +1,9 @@
 ## New Features
 
+- Update the Istio version to 1.21.0 [#676](https://github.com/kyma-project/istio/pull/676). Read [Istio 1.21.0 Release Announcement](https://istio.io/latest/news/releases/1.21.x/announcing-1.21/) and [Change Notes](https://istio.io/latest/news/releases/1.21.x/announcing-1.21/change-notes/) for more details.
 - The version details for Istio installation are now only specified in the files located in the [`istiooperator`](https://github.com/kyma-project/istio/blob/main/internal/istiooperator) directory [#664](https://github.com/kyma-project/istio/pull/664).
+
+## Breaking Changes in Istio Upstream
```
Comment (nataliasitko) [inline docs/release-notes/1.5.0.md:6]:
```suggestion
## Relevant Changes in Istio Upstream
```

**Review by nataliasitko — APPROVED** [review]

**Pull request merged**